src/Microsoft.Health.Fhir.Shared.Core/Features/Operations/Reindex/ReindexSingleResourceRequestHandler.cs

```diff
@@ -0,0 +1,93 @@
+using Microsoft.Health.Fhir.Core.Messages.Reindex;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.Reindex
+{
+    public class ReindexSingleResourceRequestHandler : IRequestHandler<ReindexSingleResourceRequest, ReindexSingleResourceResponse>
+    {
+        private readonly IFhirAuthorizationService _authorizationService;
+        private readonly IFhirDataStore _fhirDataStore;
+        private readonly ISearchIndexer _searchIndexer;
+        private readonly IResourceDeserializer _resourceDeserializer;
+
+        public ReindexSingleResourceRequestHandler(
+            IFhirAuthorizationService authorizationService,
+            IFhirDataStore fhirDataStore,
+            ISearchIndexer searchIndexer,
+            IResourceDeserializer deserializer)
+        {
+            EnsureArg.IsNotNull(authorizationService, nameof(authorizationService));
+            EnsureArg.IsNotNull(fhirDataStore, nameof(fhirDataStore));
+            EnsureArg.IsNotNull(searchIndexer, nameof(searchIndexer));
+            EnsureArg.IsNotNull(deserializer, nameof(deserializer));
+
+            _authorizationService = authorizationService;
+            _fhirDataStore = fhirDataStore;
+            _searchIndexer = searchIndexer;
+            _resourceDeserializer = deserializer;
+        }
+
+        public async Task<ReindexSingleResourceResponse> Handle(ReindexSingleResourceRequest request, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (await _authorizationService.CheckAccess(DataActions.Reindex) != DataActions.Reindex)
+            {
+                throw new UnauthorizedFhirActionException();
+            }
+
+            var key = new ResourceKey(request.ResourceType, request.ResourceId);
+            ResourceWrapper storedResource = await _fhirDataStore.GetAsync(key, cancellationToken);
+
+            if (storedResource == null)
+            {
+                throw new ResourceNotFoundException(string.Format(Core.Resources.ResourceNotFoundById, request.ResourceType, request.ResourceId));
+            }
+
+            // We need to extract the "new" search indices since the assumption is that
+            // a new search parameter has been added to the fhir server.
+            ResourceElement resourceElement = _resourceDeserializer.Deserialize(storedResource);
+            IReadOnlyCollection<SearchIndexEntry> newIndices = _searchIndexer.Extract(resourceElement);
+
+            string searchIndexString = string.Empty;
+            if (newIndices.Count > 0)
+            {
+                // A resource can have multiple values for the same search param (eg: name for Patient).
+                // Hence using a HashSet to avoid duplicate values in the response.
+                searchIndexString = string.Join(",", newIndices.Select(x => x.SearchParameter.Name).ToHashSet());
```

Comment:
It appears this will return a list of the names of the search params for the resource.  I think we should return a collection of name:value pairs, so the user can see the search param and the value that it extracted from the resource.  This will help determine if the new search param is defined correctly.

Reply:
Updated code to return search parameter name and corresponding values for the resource as part of the `Parameter` response.